tests/integration/network.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/tests/workload"
+)
+
+var (
+	ErrUnableToConfirmTx = errors.New("unable to confirm transaction")
+	ErrInvalidURI        = errors.New("invalid uri")
+	ErrTxNotFound        = errors.New("tx not found")
+)
+
+const (
+	txCheckInterval = 100 * time.Millisecond
+)
+
+type Network struct {
+	uris []string
+}
+
+func (n *Network) ConfirmTxs(ctx context.Context, uri string, txs []*chain.Transaction) error {
+	// this is the wrong way. for integration, we wante to test the block content directly.
+	instance, err := n.getInstance(uri)
+	if err != nil {
+		return err
+	}
+	expectBlk(instance)(false)
+
+	for {
+		allFound := true
+		for _, tx := range txs {
+			err := n.confirmTx(ctx, instance, tx.ID())
+			if err == ErrTxNotFound {
+				allFound = false
+				break
+			}
+			return err
```

Comment:
as mentioned before, simply returning this error is not going to be that helpful for a user